### PR TITLE
[cppcheck] fix & enable performance checks

### PR DESCRIPTION
**Pull request by Rechi**
## Description
fix & enable cppcheck performance checks

## How has this been tested?
compiled with cppcheck enabled for Android, iOS, Linux, macOS and tvOS

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by garbear — APPROVED** [review]
I approve the game change. And the rest of the commits look in order. Thanks!

File: xbmc/windowing/wayland/WinSystemWayland.cpp [hunk 1]
```diff
@@ -774,7 +774,7 @@ void CWinSystemWayland::ProcessMessages()
   }
 }
 
-void CWinSystemWayland::ApplyShellSurfaceState(IShellSurface::StateBitset state)
+void CWinSystemWayland::ApplyShellSurfaceState(const IShellSurface::StateBitset& state)
```
Comment (OBorce) [inline xbmc/windowing/wayland/WinSystemWayland.cpp:777]: from what I am seeing the StateBitset is a std:::bitset<5> which is just 8 bytes, so it should be actually cheaper to pass by value then by reference?

Comment (yol) [reply xbmc/windowing/wayland/WinSystemWayland.cpp:777]: Yeah these changes do not really seem useful

**Review by yol — CHANGES_REQUESTED** [review]

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCDDA.h [hunk 2]
```diff
@@ -41,7 +41,7 @@ class CDVDDemuxCDDA : public CDVDDemux
 
 protected:
   friend class CDemuxStreamAudioCDDA;
-  std::shared_ptr<CDVDInputStream> m_pInput;
+  std::shared_ptr<CDVDInputStream> m_pInput = nullptr;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCDDA.h:44]: No need to init, not a raw pointer.

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.h [hunk 3]
```diff
@@ -52,8 +52,8 @@ class CDVDDemuxClient : public CDVDDemux
   std::shared_ptr<CDemuxStream> GetStreamInternal(int iStreamId);
   bool IsVideoReady();
 
-  std::shared_ptr<CDVDInputStream> m_pInput;
-  std::shared_ptr<CDVDInputStream::IDemux> m_IDemux;
+  std::shared_ptr<CDVDInputStream> m_pInput = nullptr;
+  std::shared_ptr<CDVDInputStream::IDemux> m_IDemux = nullptr;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.h:56]: No need to init, not a raw pointers.

File: xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h [hunk 4]
```diff
@@ -112,7 +112,7 @@ class CDVDDemuxFFmpeg : public CDVDDemux
   bool Aborted();
 
   AVFormatContext* m_pFormatContext;
-  std::shared_ptr<CDVDInputStream> m_pInput;
+  std::shared_ptr<CDVDInputStream> m_pInput = nullptr;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h:115]: No need to init, not a raw pointer.

File: xbmc/cores/VideoPlayer/DVDMessageQueue.h [hunk 5]
```diff
@@ -19,22 +19,17 @@
...
-    priority = 0;
-  }
+  DVDMessageListItem() { priority = 0; }
   DVDMessageListItem(const DVDMessageListItem&) = delete;
   ~DVDMessageListItem() = default;
 
   DVDMessageListItem& operator=(const DVDMessageListItem&) = delete;
 
-  std::shared_ptr<CDVDMsg> message;
+  std::shared_ptr<CDVDMsg> message = nullptr;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/DVDMessageQueue.h:32]: No need to init, not a raw pointer.

File: xbmc/cores/VideoPlayer/VideoPlayerVideo.h [hunk 6]
```diff
@@ -133,7 +133,7 @@ class CVideoPlayerVideo : public CThread, public IDVDStreamPlayerVideo
   CDVDMessageQueue m_messageQueue;
   CDVDMessageQueue& m_messageParent;
   CDVDStreamInfo m_hints;
-  std::unique_ptr<CDVDVideoCodec> m_pVideoCodec;
+  std::unique_ptr<CDVDVideoCodec> m_pVideoCodec = nullptr;
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoPlayerVideo.h:136]: No need to init, not a raw pointer.

File: xbmc/cores/paplayer/VideoPlayerCodec.h [hunk 7]
```diff
@@ -42,8 +42,8 @@ class VideoPlayerCodec : public ICodec
   CAEStreamInfo::DataType GetPassthroughStreamType(AVCodecID codecId, int samplerate, int profile);
 
   CDVDDemux* m_pDemuxer;
-  std::shared_ptr<CDVDInputStream> m_pInputStream;
-  std::unique_ptr<CDVDAudioCodec> m_pAudioCodec;
+  std::shared_ptr<CDVDInputStream> m_pInputStream = nullptr;
```
Comment (ksooo) [inline xbmc/cores/paplayer/VideoPlayerCodec.h:45]: No need to init, not raw pointers.

File: xbmc/dialogs/GUIDialogKeyboardGeneric.h [hunk 8]
```diff
@@ -80,7 +80,7 @@ class CGUIDialogKeyboardGeneric : public CGUIDialog, public CGUIKeyboard
     std::string m_strHeading;
     std::string m_text;       ///< current text
 
-    IInputCodingTablePtr m_codingtable;
```
Comment (ksooo) [inline xbmc/dialogs/GUIDialogKeyboardGeneric.h:83]: Is this a raw pointer?

File: xbmc/input/KeyboardLayout.h [hunk 9]
```diff
@@ -48,5 +48,5 @@ class CKeyboardLayout
   std::string m_language;
   std::string m_layout;
   Keyboards m_keyboards;
-  IInputCodingTablePtr m_codingtable;
+  IInputCodingTablePtr m_codingtable = nullptr;
```
Comment (ksooo) [inline xbmc/input/KeyboardLayout.h:51]: Is this a raw pointer?

**Comment by Rechi** [comment]
@ksooo I've added a fixup commit which removes the default initialization with `nullptr`. I'll squash it, if evertything is okay.
@yol I've dropped the `IShellSurface::StateBitset` changes.

**Review by yol — APPROVED** [review]
wayland changes are gone, but I didn't have the time to look into the rest

**Comment by thexai** [comment]
jenkins build this please

**Review by ksooo — APPROVED** [review]
LGTM, as soon as clang-format is happy.

**Pull request merged**